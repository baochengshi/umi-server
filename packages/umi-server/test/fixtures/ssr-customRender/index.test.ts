import { join } from 'path';
import { mount } from 'enzyme';
import server from '../../../src';

describe('ssr-customRender', () => {
  it('ssr-customRender /', async () => {
    const hrstart = process.hrtime();
    const render = server({
      root: join(__dirname, 'dist'),
      customRender: async ({ htmlElement }) => {
        const wrapper = mount(htmlElement);
        await new Promise((resolve, reject) => {
          const max_time = 10 * 1000; //最大10秒
          let timeout = 0; //当前等待时间
          let interval = 300; //等待间隔
          let timer = setInterval(() => {
            wrapper.update();
            //网络请求完成
            wrapper.update();
            clearInterval(timer);
            resolve(true);
            timeout += interval;
            if (timeout >= max_time) {
              //超时
              reject(false);
            }
          }, interval);
        });
        return wrapper.html();
      },
    });

    const { ssrHtml } = await render({
      req: {
        url: '/',
      },
    });

    const [s, ms] = process.hrtime(hrstart);
    console.info('ssr-customRender / Execution time (hr): %ds %dms', s, ms / 1000000);

    expect(ssrHtml).toMatchSnapshot();
  });
});
