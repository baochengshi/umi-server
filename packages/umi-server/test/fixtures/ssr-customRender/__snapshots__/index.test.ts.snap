// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr-customRender ssr-customRender / 1`] = `"<!DOCTYPE html><html><head><link rel=\\"stylesheet\\" href=\\"/umi.css\\"><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no\\"><script>window.g_useSSR=true;</script><script>window.routerBase = \\"/\\";</script></head><body><div id=\\"root\\"><div class=\\"wrapper\\"><h1>Hello UmiJS SSR Styles</h1><ul><li>Alice</li><li>Jack</li><li>Tony</li></ul></div></div><script>window.g_initialData = {};</script><script src=\\"/umi.js\\"></script></body></html>"`;
